sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3406,9 +3413,12 @@ def teardown_no_lock(self,
                     query_cmd = (f'gcloud compute instances list --filter='
                                  f'"(labels.ray-cluster-name={cluster_name})" '
                                  f'--zones={zone} --format=value\\(name\\)')
+                    # If there are no instances, exit with 0 rather than causing
+                    # the delete command to fail.
                     terminate_cmd = (
-                        f'gcloud compute instances delete --zone={zone}'
-                        f' --quiet $({query_cmd})')
+                        f'NAMES=$({query_cmd}) && [ -n "$NAMES" ] && '
+                        f'gcloud compute instances delete --zone={zone} --quiet'
+                        ' $NAMES || echo "No instances to delete."')
```

Comment:
Nice! Can we also make the AWS termination command robust, and set the L3388 to be `prev_cluster_status != status_lib.ClusterStatus.UP` to fix #2089. We can do it in another PR as well.

Reply:
I added the empty-list handling to the AWS CLI invocation. Will leave #2089 to later which seems to require more involved handling (down the head node first).